routers/web/web.go

```diff
@@ -644,7 +645,18 @@ func registerRoutes(m *web.Route) {
 			m.Get("", admin.RedirectToDefaultSetting)
 			addSettingsRunnersRoutes()
 		})
-	}, adminReq, ctxDataSet("EnableOAuth2", setting.OAuth2.Enable, "EnablePackages", setting.Packages.Enabled))
+	}, adminReq, ctxDataSet(
+		"EnableOAuth2", setting.OAuth2.Enable,
+		"EnablePackages", setting.Packages.Enabled,
+	), func(ctx *context.Context) {
+		if len(setting.DeprecatedWarnings) > 0 {
+			content := setting.DeprecatedWarnings[0].String()
+			if len(setting.DeprecatedWarnings) > 1 {
+				content += fmt.Sprintf(" (and %d more)", len(setting.DeprecatedWarnings)-1)
+			}
+			ctx.Flash.Error(content)
+		}
+	})
```

Comment:
I would like to see this in a method instead inlined.

Reply:
done.